src/content/app/genome-browser/components/track-panel/components/track-panel-modal/TrackPanelModal.tsx

```diff
@@ -14,58 +14,74 @@
-import TracksManager from './modal-views/TracksManager';
-import TrackPanelBookmarks from './modal-views/TrackPanelBookmarks';
-import PersonalData from './modal-views/PersonalData';
-import TrackPanelShare from './modal-views/TrackPanelShare';
-import TrackPanelDownloads from './modal-views/TrackPanelDownloads';
-
 import SidebarModal from 'src/shared/components/layout/sidebar-modal/SidebarModal';
 
+import { TrackPanelModalView } from 'src/content/app/genome-browser/state/track-panel/trackPanelSlice';
+
+const trackPanelSidebarModals: Record<
+  string,
+  LazyExoticComponent<() => JSX.Element | null>
+> = {
+  [TrackPanelModalView.SEARCH]: lazy(
+    () => import('./modal-views/TrackPanelSearch')
+  ),
+  [TrackPanelModalView.TRACKS_MANAGER]: lazy(
+    () => import('./modal-views/TracksManager')
+  ),
+  [TrackPanelModalView.BOOKMARKS]: lazy(
+    () => import('./modal-views/TrackPanelBookmarks')
+  ),
+  [TrackPanelModalView.PERSONAL_DATA]: lazy(
+    () => import('./modal-views/PersonalData')
+  ),
+  [TrackPanelModalView.SHARE]: lazy(
+    () => import('./modal-views/TrackPanelShare')
+  ),
+  [TrackPanelModalView.DOWNLOADS]: lazy(
+    () => import('./modal-views/TrackPanelDownloads')
+  )
+};
+
+export const trackPanelModalTitles: { [key: string]: string } = {
+  [TrackPanelModalView.SEARCH]: 'Search',
+  [TrackPanelModalView.TRACKS_MANAGER]: 'Tracks manager',
+  [TrackPanelModalView.BOOKMARKS]: 'Previously viewed',
+  [TrackPanelModalView.PERSONAL_DATA]: 'Personal data',
+  [TrackPanelModalView.SHARE]: 'Share',
+  [TrackPanelModalView.DOWNLOADS]: 'Downloads'
+};
+
 export const TrackPanelModal = () => {
   const trackPanelModalView = useSelector(getTrackPanelModalView);
   const dispatch = useDispatch();
 
-  const getModalViewData = () => {
-    switch (trackPanelModalView) {
-      case 'search':
-        return { content: <TrackPanelSearch />, title: 'Search' };
-      case 'tracks-manager':
-        return { content: <TracksManager />, title: 'Tracks manager' };
-      case 'bookmarks':
-        return { content: <TrackPanelBookmarks />, title: 'Previously viewed' };
-      case 'personal-data':
-        return { content: <PersonalData />, title: 'Personal data' };
-      case 'share':
-        return { content: <TrackPanelShare />, title: 'Share' };
-      case 'downloads':
-        return { content: <TrackPanelDownloads />, title: 'Downloads' };
-      default:
-        return {
-          content: null,
-          title: ''
-        };
-    }
-  };
+  if (!trackPanelModalView) {
+    return null;
+  }
+
+  const ModalView = trackPanelSidebarModals[trackPanelModalView];
+  const modalViewTitle = trackPanelModalTitles[trackPanelModalView];
 
   const onClose = () => {
     dispatch(closeDrawer());
     dispatch(closeTrackPanelModal());
   };
 
-  const { title, content } = getModalViewData();
   return (
-    <SidebarModal title={title} onClose={onClose}>
-      {content}
-    </SidebarModal>
+    <section>
```

Comment:
I'm unsure of this html element. At least in the EntityViewer it is trying to do some useful job:

![image](https://user-images.githubusercontent.com/6834224/161925247-b32a6dd6-1224-43df-b029-6f0d0b9f44d8.png)

whereas here it isn't doing anything useful.

We might want to replicate here what the section element is doing in the EntityViewer (height 100%, overflow: auto). Or, failing that, get rid of this element altogether.

Reply:
I've removed the `section` element from `TrackPanelModal`. Also removed it from `EntityViewerSidebarModal`. None of the styling in `EntityViewerSidebarModal.scss` was used either so deleted the file.